src/openfermion/ops/_polynomial_tensor.py

```diff
@@ -25,10 +25,57 @@ class PolynomialTensorError(Exception):
     pass
 
 
+def general_basis_change(general_tensor, rotation_matrix, key):
+    """Change the basis of an general interaction tensor.
+
+    M' = R^T.M.R where R is the rotation matrix, M is the general tensor
+    and M' is the transformed general tensor.
+
+    Args:
+        general_tensor: A square numpy array or matrix containing information
+            about a general interaction tensor.
+        rotation_matrix: A square numpy array or matrix having dimensions of
+            n_qubits by n_qubits. Assumed to be unitary.
+        key: A tuple indicating the type of general_tensor. Assumed to be
+            non-empty.
+
+    Returns:
+        transformed_general_tensor: general_tensor in the rotated basis.
+    """
+    # If operator acts on spin degrees of freedom, enlarge rotation matrix.
+    n_orbitals = rotation_matrix.shape[0]
+    if general_tensor.shape[0] == 2 * n_orbitals:
+        rotation_matrix = numpy.kron(rotation_matrix, numpy.eye(2))
+
+    order = len(key)
+    if order > 26:
+        raise PolynomialTensorError('Order exceeds maximum order '
+                                    'supported (26).')
+
+    # 'abcd'
+    subscripts_first = ''.join(chr(ord('a') + i) for i in range(order))
+
+    # 'aA,bB,cC,dD'
+    subscripts_rest = ','.join(chr(ord('A') + i) +
+                               chr(ord('a') + i) for i in range(order))
+
+    subscripts = subscripts_first + ',' + subscripts_rest
+
+    rotation_matrices = [rotation_matrix.conj() if x == 1 else
+                         rotation_matrix for x in key]
+
+    # "optimize = True" does greedy optimization, which will be enough here
+    transformed_general_tensor = numpy.einsum(subscripts,
+                                              general_tensor,
+                                              *rotation_matrices,
```

Comment:
What does the `*` in `*rotation_matrices` do?

Reply:
It's a Python syntax to pass a variable number of parameters to a function. For example, if `key` is `(0,0,0)` then this is equivalent to `numpy.einsum(subscripts,  general_tensor, rotation_matrix, rotation_matrix, rotation_matrix)`.

Reply:
Cool!